ts/DataGrid/DataGrid.ts

```diff
@@ -937,6 +943,55 @@ class DataGrid {
         DataGrid.dataGrids.splice(dgIndex, 1);
     }
 
+    /**
+     * Grey out the data grid and show a loading indicator.
+     *
+     * @param message
+     * The message to display in the loading indicator.
+     */
+    public showLoading(message?: string): void {
+        // Create loading wrapper.
+        this.loadingWrapper = makeHTMLElement(
+            'div',
+            {
+                className: 'highcharts-datagrid-loading-wrapper',
+            },
+            this.contentWrapper
+        )
```

Comment:
Before showing the loading, it would be useful to check whether the loading state is not active in the moment, to prevent `loadingWrapper` HTML Elements from overlapping if the function is called multiple times.

You can reproduce this by repeatedly clicking fast on the `Load data` button in the sample you added.